### PR TITLE
Allow longhand and shorthand properties in `theme.json` and block attributes

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Related: https://github.com/WordPress/gutenberg/issues/31337, https://github.com/WordPress/gutenberg/pull/31585
~~Depends on: https://github.com/WordPress/gutenberg/pull/31483~~

## Description

This PR:
- Flattens the [configuration array used to map CSS property names](https://github.com/WordPress/gutenberg/pull/31641/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R198) with paths to the appropriate value in theme.json and block attributes
- Updates border and spacing support so that border radius, margin and padding support allow both shorthand and longhand styles e.g. `margin: 10px` or `margin-left: 20px`
- Updates the theme.json and block support style tests to cover shorthand and longhand styles
- Updates global styles renderer to handle shorthand/longhand style values and configuration

The driving force behind this was achieving the refined border support UI in https://github.com/WordPress/gutenberg/issues/31337. To do so, the border block support needs to allow custom values for each corner's border radius. This PR addresses this need while also maintaining backwards compatibility for a single border radius value.

## How has this been tested?
Automated tests.

Theme.json stylesheet generation: 
`/var/www/html/wp-content/plugins/gutenberg/phpunit/class-wp-theme-json-test.php`

Block support inline styles: 
`packages/block-editor/src/hooks/test/style.js`
`packages/edit-site/src/components/editor/test/global-styles-renderer.js`

Manual testing:
1. Apply this PR
2. Enable custom margin and padding support via theme.json
3. Add theme styles for the `core/group` block in the theme.json as well. Try simple shorthand margin and padding values
4. Open the site editor and in the global styles sidebar switch to the `By block type` tab, then expand the Group block panel.
5. Confirm the margin and padding controls work and their global style values can be updated and saved correctly
6. Edit the theme.json margin / padding styles to use split values per side
7. Reload the site editor and confirm the group block's margin/padding controls still function
8. Create a post within the normal block editor and add a group block
9. Select the group block and configure its margin and paddding
10. Save and confirm the spacing displays correctly.

_This PR does not update the border support UI, to test the border changes manually, try https://github.com/WordPress/gutenberg/pull/31585_

1. Checkout out the #31585 branch `update/border-block-support-ui`
    (that's all that is needed, it contains the changes from this PR)
2. Enable border support UI via theme.json - [example theme.json gist](https://gist.github.com/aaronrobertshaw/4deded861032467d4dccfe0797d3befd)
3. Build (you might need to run `npm install`)
4. Load editor, create post, add group block, and select it
5. Confirm border support panel and radius control display and function as expected

## Screenshots 
Via use of these changes in https://github.com/WordPress/gutenberg/pull/31585

## Types of changes
New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
A few wee early and very initial thoughts: the border radius panel should probably start collapsed and default to PX rather than EM. I'd also put color below both border style, width and radius. 

We also really need the collapsed color swatch from #27331 and https://github.com/WordPress/gutenberg/issues/27473#issuecomment-796773866 soon, it simply takes up too much space in this vein. 

Honestly I wish we could collapse the border control as far down as this:

Screenshot 2021-05-11 at 08 59 27

— that is, slightly Figma inspired, a border is something you _add_: 

border

We'd need for the border controls to be in a separate panel that could look like #31337, right @jameskoster?

**Comment by aaronrobertshaw** [comment]
Thanks for the early feedback @jasmussen! It's very much appreciated.

> A few wee early and very initial thoughts: the border radius panel should probably start collapsed and default to PX rather than EM. 

The screenshot in the PR description was mistakenly captured after I'd already set border radii on the block while testing. That's why the border radius field defaulted to open, I'd set specific per-corner values for it. If you were testing via https://github.com/WordPress/gutenberg/pull/31585 this is what I see:



> I'd also put color below both border style, width and radius.

The current order (in https://github.com/WordPress/gutenberg/pull/31585) matches the order of controls in https://github.com/WordPress/gutenberg/issues/31337. I believe the idea there was for them to match the CSS syntax order. Can you confirm I should change the order so the color controls appear last?

> We also really need the collapsed color swatch from #27331 and #27473 (comment) soon, it simply takes up too much space in this vein.

I can appreciate the desire for the collapsed color swatch. I was also looking at being able to [add separate background & text color controls for table ](https://github.com/WordPress/gutenberg/issues/31261) header, footer and striped rows as requested in https://github.com/WordPress/gutenberg/pull/30791#issuecomment-823815094. The current control is definitely too unwieldy to provide that many options.

Screen Shot 2021-05-11 at 5 18 36 pm

Looking at the initial designs for the collapsed color control, it looks to me like it will take quite some time to be able to implement along with required subcomponents. 

I don't think it should hold up the other refinements to the border block support UI ( https://github.com/WordPress/gutenberg/issues/31337 ) as there are several PRs blocked waiting to be able to leverage this support. In addition, the controls are disabled by default which might provide the window to iterate on the color control.

I'm happy to make a start on implementing the collapsed color control as soon as I free up a little extra bandwidth.

> We'd need for the border controls to be in a separate panel that could look like #31337, right @jameskoster?

I might be missing something but this is what https://github.com/WordPress/gutenberg/pull/31585 is looking to achieve. However, to achieve some of the controls in that UI design we needed to refine the border block support to [handle non-pixel units](https://github.com/WordPress/gutenberg/pull/31483) and the split border radius values in this PR.

**Comment by jameskoster** [comment]
> Honestly I wish we could collapse the border control as far down as this:

The missing piece there is being able to toggle the _panels_ themselves in the UI. I don't think we ever got to designs for that in https://github.com/WordPress/gutenberg/issues/27331.

@aaronrobertshaw This is looking good from the screenshots, but I'm not seeing the controls in my local environment. Is there anything I need to do besides building this branch? 

One other question: were the icons for the border style added to the icons package?

**Comment by aaronrobertshaw** [comment]
Thanks for testing @jameskoster 👍 

> This is looking good from the screenshots, but I'm not seeing the controls in my local environment. 

This PR itself only updates the block support etc to be capable of handling the individual border radius values. It includes updates to automated tests that can be run.

> Is there anything I need to do besides building this branch?

I have a separate PR ( https://github.com/WordPress/gutenberg/pull/31585 ) based upon this one that introduces the new border radius control and refines the border block support UI. It's currently only a draft and marked as WIP as I need to confirm the approach taken both in this PR and for custom units for border width and radius.

It would be great if you still had time to give this all a test. To do that, build https://github.com/WordPress/gutenberg/pull/31585 and then enable custom borders in your `experimental-theme.json`. The group block already opts into all the border block support features so it's the easiest block to add to a post and test with.

> One other question: were the icons for the border style added to the icons package?

Not at this stage, they were packaged with the new border radius control. I was suffering from some tunnel vision while putting together the PR refining the border support UI. I can add them to the icons package as part of https://github.com/WordPress/gutenberg/pull/31585 or a separate PR, whichever is best.

Also, from the early discussion on that border radius control, it sounded as though it would need its own icon to represent corners rather than sides for the linked button. However, if the simple link/unlink icons used at the moment are ok, it might be possible to iterate on that control and make it more generic. Something we can work on via a separate PR.

**Comment by jameskoster** [comment]
> enable custom borders in your experimental-theme.json

Asking for a friend... how does one do this? 😆

**Comment by aaronrobertshaw** [comment]
> Asking for a friend... how does one do this? 😆

Sorry @jameskoster I hadn't gotten to detailing test instructions on the WIP PR. The easiest method for testing is to just toggle on the custom border properties under the `defaults` settings in the theme.json. Below is a snippet of the relevant config flags or for more context a [gist of my default theme.json with custom borders](https://gist.github.com/aaronrobertshaw/4deded861032467d4dccfe0797d3befd) enabled

```
		"border": {
			"customColor": true,
			"customRadius": true,
			"customStyle": true,
			"customWidth": true
		},
```

**Comment by jameskoster** [comment]
I must be doing something wrong 🙈 

> To do that, build #31585 and then enable custom borders in your experimental-theme.json.

I've done this and I'm not seeing border controls anywhere, even on the group block.

**Comment by aaronrobertshaw** [comment]
> I must be doing something wrong 🙈

In this case, it's most likely poor instructions 🙂 

I've retested and after checking out https://github.com/WordPress/gutenberg/pull/31585 and toggling all the border properties below to `true`. The border support panel shows for me on the group block.

```json
		"border": {
			"customColor": true,
			"customRadius": true,
			"customStyle": true,
			"customWidth": true
		},
```

I think the problem could be that I linked to a gist that had borders enabled but was more recent. The theme.json schema has been recently reshaped. It caught me out on a couple of PRs. If you copied and pasted the entire gist it wouldn't have matched the schema and been misinterpreted.

Here is a [gist](https://gist.github.com/aaronrobertshaw/4deded861032467d4dccfe0797d3befd) you can copy and paste into your theme's `experimental-theme.json` file or the `lib/experimental-default-theme.json` if appropriate. I've tested copying and pasting this gist into my theme.json and it works.

> One other question: were the icons for the border style added to the icons package?

The icons have been added to the icons package via a [commit](https://github.com/WordPress/gutenberg/pull/31585/commits/642806870807428e766a3f5677fbeb4538bdfb12) in https://github.com/WordPress/gutenberg/pull/31585.

**Comment by jameskoster** [comment]
No luck :( FWIW I'm using tt1-blocks which seems to have the correct declaration: https://github.com/WordPress/theme-experiments/blob/master/tt1-blocks/theme.json#L188-L193

**Comment by ramonjd** [comment]
I'm also having trouble testing this manually. 

I applied this patch on top of https://github.com/WordPress/gutenberg/pull/31483 and set the following in `settings.default` in `lib/experimental-default-theme.json`

```
			"border": {
				"customColor": true,
				"customRadius": true,
				"customStyle": true,
				"customWidth": true
			}
```

Should `customRadius` be an array of `[ "topLeft", ... ]` or something?

I tried a few themes and seeing these controls 

Screen Shot 2021-05-28 at 3 34 32 pm

Are there updated instructions?

**Comment by ramonjd** [comment]
> No luck :( 

@jameskoster I think we have to test these changes manually by checking out this PR branch https://github.com/WordPress/gutenberg/pull/31585

> Automated tests.

Unit tests passed locally for me 👍

**Comment by aaronrobertshaw** [comment]
> I'm also having trouble testing this manually.

Thanks @ramonjd for testing.

I'd tried to separate the changes related to updating the border radius support from the updates to the border support UI. The idea being to keep the PRs a manageable size. That's why I only had instructions related to the automated tests on this PR. Perhaps adding the gif illustrating what this PR allows caused some confusion. My apologies.

> Are there updated instructions?

I'll update the test instructions for those that would like to see the updated block support in action. To test this support manually, we need the updated border support UI in https://github.com/WordPress/gutenberg/pull/31585.

1. Checkout out the https://github.com/WordPress/gutenberg/pull/31585 branch `update/border-block-support-ui` 
    (that's all that is needed, it contains the changes from this PR)
2. Enable border support UI via theme.json - [example theme.json gist](https://gist.github.com/aaronrobertshaw/4deded861032467d4dccfe0797d3befd)
3. Build
3. Load editor, create post, add group block, and select it
4. Confirm border support panel and radius control display and function as expected

cc/ @jameskoster in case this helps.

**Comment by ramonjd** [comment]
> That's why I only had instructions related to the automated tests on this PR.

Thanks @aaronrobertshaw It's totally obvious now you point it out. :D I think I was fixated on the GIF and design feedback and assumed...

**Comment by aaronrobertshaw** [comment]
> It's totally obvious now you point it out. :D I think I was fixated on the GIF and design feedback and assumed...

I appreciate the feedback all the same. I'll try and make the distinction clearer on future PRs. The little comment above the gif "Via use of these changes in #31585" is pretty easy to scan straight passed and isn't exactly clear either.

Thanks again for testing 👍 

I've also re-tested using the new instructions and it is still working for me.

File: lib/class-wp-theme-json-gutenberg.php
Comment (youknowriad) [inline lib/class-wp-theme-json-gutenberg.php]: quick question here for @nosolosw @jorgefilipecosta what happens if you run this on top of `wp/trunk` I mean will the class from Gutenberg be used or the one from Core? I guess it should be Gutenberg's with the updates but just wanted to confirm.

File: lib/class-wp-theme-json-gutenberg.php
Comment (youknowriad) [inline lib/class-wp-theme-json-gutenberg.php]: @aaronrobertshaw Why is it different than `margin`/`padding` For these as well e need to support `margin: 10px` as well, no?

File: packages/blocks/src/api/constants.js
Comment (youknowriad) [inline packages/blocks/src/api/constants.js:36]: The config here looks also different than margin/padding and I'm not certain why it diverged?

File: lib/class-wp-theme-json-gutenberg.php
Comment (aaronrobertshaw) [reply lib/class-wp-theme-json-gutenberg.php]: I'm not sure I follow your question sorry. Hopefully the following information clarifies things.

For margin/padding, the normal approach takes the key from this metadata array and then adds the individual sides as a suffix to build the CSS property. For example, `margin` with the properties array of `[ 'left', 'right']` would currently create two CSS properties `margin-left` and `margin-right`. 

The border radius CSS properties for individual corners follow a different format than the current approach for margins & paddings. These CSS properties would need the properties array values to be infixed within the normal `border-radius` key. I felt there was less "magic" going on if the CSS property names could actually just be explicitly set if needed. This also helps for [possible new block supports such as grid gaps](https://github.com/WordPress/gutenberg/pull/32571) where the CSS property name is a different format again.

The original `border-radius` entry in the properties metadata array remains for backwards compatibility. It functions as normal (when there isn't per-corner configuration for border radii). The CSS generation here and in global styles, as well as the block support inline styles, check whether the  properties value is an object and if so uses the keys as the CSS property name rather than generating one from a formula.

I'm definitely open to better ideas. This was all that I came up with that would work.

File: packages/blocks/src/api/constants.js
Comment (aaronrobertshaw) [reply packages/blocks/src/api/constants.js:36]: Yes, this configuration is an object on purpose. It allows for the differentiation between styles that are to generate their CSS property name via the generic formula `<key>-<property>` and those for which that approach doesn't work e.g. `border-top-left-radius` and `row-gap`. 

Having the `properties` property also support an object if needed, allows that object's keys to be used as explicit CSS property names which covers the atypical ones mentioned above.

I took this approach as it had the least impact on the ways styles were generated previously but expanded what was possible.

Again, open to suggestions here.

File: lib/class-wp-theme-json-gutenberg.php
Comment (youknowriad) [reply lib/class-wp-theme-json-gutenberg.php]: > The original border-radius entry in the properties metadata array remains for backwards compatibility. 

Can you clarify how could this entry be used by third-party devs? (just trying to understand the impact).

--- 
Thanks for the explanation, it makes sense but it seems that your solution is the most generic one and I wonder whether we should just use it for padding/margin to stay consistent?

Also cc @nosolosw @jorgefilipecosta as they know more about these APIs...

Comment (aaronrobertshaw) [reply lib/class-wp-theme-json-gutenberg.php]: > Can you clarify how could this entry be used by third-party devs? (just trying to understand the impact).

The idea was that the border radius could be specified via the normal `border.radius` property within the theme.json so there would be no change for third-party devs unless I'm missing something.

When the properties metadata is processed into style declarations both configuration entries would have the same value path. A check for whether the value was a string or not allowed the correct css rule to be generated.

> Thanks for the explanation, it makes sense but it seems that your solution is the most generic one and I wonder whether we should just use it for padding/margin to stay consistent?

Ah, I think I'm following now, thanks for the extra questions.

I spent some time this afternoon to try out consolidating the properties metadata entries into a single entry each for border radius, margin and padding. As part of this I've made additional updates so that those single entries can support both shorthand and longhand values e.g. `margin: 10px`, `margin-left: 10px` or `border-top-left-radius: 2px`.

After doing this the existing tests were all passing along with new tests to ensure both shorthand and longhand rule generation worked. It could do with a bit of an extra test before publishing as I'm not certain how the possibility of a shorthand value from the theme.json will work with the Global Styles UI using a box control expecting split values. I'll take a look at this aspect tomorrow.

Will it be OK to push these new changes to this PR, so the configuration updates are all together and with the discussion around why they are required?

Comment (oandregal) [reply lib/class-wp-theme-json-gutenberg.php]: It looks like we can simplify the existing code in `trunk` if we take one of the ideas proposed in this PR and push it further: make the `PROPERTIES_METADATA` a direct mapping from the CSS property name and the path where that property is stored in `theme.json`.

```php
const PROPERTIES_METADATA = array(
	'background-color'           => array( 'color', 'background' ),
	'background'                 => array( 'color', 'gradient' ),
	'border-color'               => array( 'border', 'color' ),
	'border-radius'              => array( 'border', 'radius' ),
	'border-top-left-radius'     => array( 'border', 'radius', 'topLeft' ),
	'border-top-right-radius'    => array( 'border', 'radius', 'topRight' ),
	'border-bottom-right-radius' => array( 'border', 'radius', 'bottomRight' ),
	'border-bottom-left-radius'  => array( 'border', 'radius', 'bottomLeft' ),
	'border-style'               => array( 'border', 'style' ),
	'border-width'               => array( 'border', 'width' ),
	'color'                      => array( 'color', 'text' ),
	'font-family'                => array( 'typography', 'fontFamily' ),
	'font-size'                  => array( 'typography', 'fontSize' ),
	'font-style'                 => array( 'typography', 'fontStyle' ),
	'font-weight'                => array( 'typography', 'fontWeight' ),
	'letter-spacing'             => array( 'typography', 'letterSpacing' ),
	'line-height'                => array( 'typography', 'lineHeight' ),
	'margin'                     => array( 'spacing', 'margin' ),
	'margin-top'                 => array( 'spacing', 'margin', 'top' ),
	'margin-right'               => array( 'spacing', 'margin', 'right' ),
	'margin-bottom'              => array( 'spacing', 'margin', 'bottom' ),
	'margin-left'                => array( 'spacing', 'margin', 'left' ),
	'padding'                    => array( 'spacing', 'padding' ),
	'padding-top'                => array( 'spacing', 'padding', 'top' ),
	'padding-right'              => array( 'spacing', 'padding', 'right' ),
	'padding-bottom'             => array( 'spacing', 'padding', 'bottom' ),
	'padding-left'               => array( 'spacing', 'padding', 'left' ),
	'text-decoration'            => array( 'typography', 'textDecoration' ),
	'text-transform'             => array( 'typography', 'textTransform' ),
);
```

The other idea this PR introduces is to allow theme authors to either declare a shorthand (`border-radius`) or longhand (`border-top-left-radius`) via `theme.json` (either `border.radius` or `border.radius.topLeft` would work now). I think this is also nice and something `margin` and `padding` would benefit as well.

If it becomes convoluted to do the two things at once we could land first a PR that just changes the implementation but doesn't change behavior, and a second one after that adds support for both long and shorthands. I'm fine reviewing it altogether now that I'm more familiar with these changes ― I also appreciate that UI changes have been already moved to a different PR.

File: lib/class-wp-theme-json-gutenberg.php
Comment (oandregal) [reply lib/class-wp-theme-json-gutenberg.php]: The plugin's code overwrites core's. In the front-end by hooking into the existing `wp_styles` ([code](https://github.com/WordPress/gutenberg/blob/trunk/lib/global-styles.php#L65)); in the editor by using the `__unstableType=globalStyles` ([code](https://github.com/WordPress/gutenberg/blob/trunk/lib/global-styles.php#L137)).

File: lib/class-wp-theme-json-gutenberg.php
Comment (aaronrobertshaw) [reply lib/class-wp-theme-json-gutenberg.php]: Thanks @nosolosw.

I've pushed changes that flatten the `PROPERTIES_METADATA` array and update where that is used accordingly. It definitely simplified some of the code in `class-wp-theme-json-gutenberg.php`. Tests have been updated to cover shorthand and longhand styles as well. 

The `STYLE_PROPERTY` configuration in `packages/blocks/src/api/constants.js` didn't really appear to gain as much if flattened, given its inclusion of the path to check support as well. In the end, I didn't flatten that config, only update the margin and padding support to be consistent with the explicit CSS property naming approach. That itself saved a few conditionals in `style.js` etc.

After pushing the initial changes, I found that I did need to tweak the spacing panel for the Global Styles sidebar so it handled a theme set value that could be either a shorthand value or array representing each side. Nothing too major.

Now the extent of changes required is known, I'm happy to split this PR if it is too much.

The PR description has been updated to reflect the current nature of changes included in this PR.

File: packages/block-editor/src/hooks/style.js
Comment (oandregal) [inline packages/block-editor/src/hooks/style.js]: Can it be the case that the block attributes only contain a subset of the longhand values (e.g.: margin-top, margin-bottom)? If so, we'd need to cover against that (not creating an inline style if a longhand such as margin-right doesn't exist).

File: lib/class-wp-theme-json-gutenberg.php
Comment (oandregal) [reply lib/class-wp-theme-json-gutenberg.php]: Thanks! It's a lot cleaner now. We can remove the `has_properties` method as well, it's no longer used.

File: lib/block-supports/border.php
Comment (oandregal) [inline lib/block-supports/border.php:68]: Can the border radius be `null`? Do we need to check for that like we do for spacing?

**Comment by oandregal** [comment]
There seems to be some conflict in files and I've left some minor comments. I think once that's addressed we can merge this one (I can manually test then, haven't done yet).

File: packages/block-editor/src/hooks/style.js
Comment (aaronrobertshaw) [reply packages/block-editor/src/hooks/style.js]: The block attributes can contain only a subset of the longhand values as you described. 

The code prior to this PR, created inline styles for all the defined subpaths as well. I believe they are just omitted when the inline styles are applied. Thankfully, that means we can tighten up this code without causing block validation errors. 

I've updated this in https://github.com/WordPress/gutenberg/pull/31641/commits/706321be7b993e8223f141e7b9573d11d745af4f

File: lib/block-supports/border.php
Comment (aaronrobertshaw) [reply lib/block-supports/border.php:68]: I believe the original code here for the border support was added prior to `_wp_array_get` getting wider usage so the code approaches the check a little differently to the spacing support. 

Along with checking for the feature support, it also does an `isset` check which would return false for a null value. Would making the supports' code consistent in approach be best done in a separate PR? I'm happy to take care of that in a follow up.

**Comment by aaronrobertshaw** [comment]
> There seems to be some conflict in files and I've left some minor comments. I think once that's addressed we can merge this one (I can manually test then, haven't done yet).

The conflicts turned out to be only for the assertions relating to generated stylesheets. I've resolved those and the tests are all green again. I'll rebase the PR updating the border support UI onto this shortly so it can be used for manual testing if desired.

**Review by oandregal — APPROVED** [review]
Tested this using padding -both short and longhands- in the front, site, and post editor. Also tested that content created in `trunk` works well with this PR (no invalidations, etc).

Thanks to your patience through this we're getting a great simplification in the implementation + new behaviour across the board. :star2: 

I'm just crossing my fingers for this not to cause too many headaches when/if we need to port late things to core for 5.8 :sweat_smile: :crossed_fingers: :four_leaf_clover: I'll take care of that should it happen.

File: packages/block-editor/src/hooks/style.js
Comment (oandregal) [reply packages/block-editor/src/hooks/style.js]: I've tested this and the UI controls don't allow to set only some of the longhands (pre-existing behavior): they'll need to be null or will be reset to the previous value. I've tested this by tweaking the block markup manually -to simulate a situation in which the UI control allows you to do that- and it works fine.

File: lib/compat.php
Comment (youknowriad) [inline lib/compat.php:188]: Was this change backported to Core at any point?

Comment (aaronrobertshaw) [reply lib/compat.php:188]: It appears to me that the [changes were backported](https://github.com/WordPress/wordpress-develop/blob/5eede7436a7938593bdc02b8653aee794b2aee63/src/wp-includes/kses.php#L2317-L2334) and were done so in (https://github.com/WordPress/wordpress-develop/commit/f034bc832eb091079e4130f6a206b4f496766796).

Comment (youknowriad) [reply lib/compat.php:188]: Part of my confusion is that the function indicates that it should be removed once the minimum WP version is 5.8, I guess it should have been updated to 5.9 after this change.

Comment (aaronrobertshaw) [reply lib/compat.php:188]: That was an oversight of mine sorry. This change didn't make it into a release until 11.0 and therefore WP 5.9. 

An additional attribute was added to this filter a few months later (https://github.com/WordPress/gutenberg/pull/36280) that would also make the minimum version requirement 5.9 before it can be removed. 

I've created a tiny PR to fix this comment in https://github.com/WordPress/gutenberg/pull/38359.

**Pull request merged**
